components/fxa-client/src/http_client/browser_id.rs

```diff
@@ -215,7 +231,7 @@ fn get_oauth_audience(oauth_url: &Url) -> Result<String> {
     }
 }
 
-fn derive_key_from_session_token(session_token: &[u8]) -> Result<Vec<u8>> {
+pub(crate) fn derive_key_from_session_token(session_token: &[u8]) -> Result<Vec<u8>> {
```

Comment:
I know this is an existing function, but I found the name of it quite confusing, because I interpreted `key` as "encrpytion key" whereas I these are the hawk authentication keys. Perhaps we could name this something like `derive_hawk_auth_key_from_session_token` or similar to make it a bit clearer?

Reply:
(Or, perhaps we could make the `HawkRequestBuilder` abstraction take the string "sessionToken" and derive the necessary keys internally)

Reply:
Ah totally agree with the confusion, I remember starring at the name of this :)